pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -772,6 +665,24 @@ func (p *PriorityQueue) newPodInfo(pod *v1.Pod) *podInfo {
 	}
 }
 
+// GetUnschedulableOrBackoffPod returns the the error Pod if it is available in the
+// unschedulableQ or podBackoffQ. This function currently is only for test purpose.
+func (p *PriorityQueue) GetUnschedulableOrBackoffPod(pod *v1.Pod) (*v1.Pod, bool, error) {
```

Comment:
Since this function is used only for tests, we should add it to the unit-test file, probably `scheduling_queue_test.go`. I know that factory_test.go also uses it, but I think only unit-tests (tests in scheduling_queue_test.go) should check which internal queue (active, unschedulable, or back-off queue) a pod is in. Any test outside of the module should only check whether a pod is in the queue or not. If those tests check internal queues, we should probably bring the tests to scheduling_queue_test.go. `PendingPods()` methods of the queue already gives you the list of pods which are in the queue. That should be enough for tests outside of the module.

Reply:
Thanks for the suggestions. I didn't notice PendingPods(). I will add a filter function in factor_test.go to find the target pod from the pending pod list. 

Since the added function is only served for factory_test.go, it can be removed from the changes for scheduling_queue.go and its unit test file.

Reply:
Updated the above changes in [commit:edc2e22](https://github.com/kubernetes/kubernetes/pull/76301/commits/edc2e22489de576ce254c5a1bf8dc213b2c4a951) 
- add a function getPodfromPriorityQueue() in factory_test.go to iterate the pod list got from PendingPods()
- rewrite TestDefaultErrorFunc() to avoid duplicate loops of testing client get action for the two cases of adding pods to backoff queue or unschedulable queue.
- remove GetUnschedulableOrBackoffPod() from scheduling_queue.go and its unit tests